src/components/CarContainer.component.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+import { CarRowInfo } from "../interfaces/Car";
+import { IMG_URL } from "../utils/index";
+import { FlexColumn, H3, ValidationText } from "./styled";
+import { CarInfo } from "./CarInfo.component";
+import { ApolloError } from "@apollo/client";
+import Loader from "./styled/Loader.component";
+
+interface Props {
+  data: {
+    cars: CarRowInfo[] | null;
+    error: ApolloError | undefined;
+    loading: boolean;
+  };
+}
+
+export const CarContainer = ({ data }: Props) => {
+  const { cars, error, loading } = data;
+  const [renderedCars, setRenderedCars] = useState<CarRowInfo[] | null>(null);
+
+  useEffect(() => {
+    setRenderedCars(cars);
+  }, [cars]);
+
+  return (
+    <Container>
+      {loading ? (
+        <>
+          <Loader />
+        </>
+      ) : error ? (
+        <ValidationText>Couldn't load data - Try again later</ValidationText>
+      ) : renderedCars?.length ? (
```

Comment:
I suggest to use a simple if to validate different states of the component instead of having complex ternary operators, this makes the code less readable

Reply:
I handled this in the latest commit using a different logic, imo looks better now, thanks!